### DISCUSSION TITLE
Pagination for External REST API Datasource

### DISCUSSION THREAD

**Discussion by lollococce** [Budibase/budibase#5026] [Help] [answered]
Hi!

Am not sure if and how I can take advantage of the pagination built in tool for the REST API with the Facebook API source.

My API call https://graph.facebook.com/v{{ version }}/act_{{ account_id }}/ads?access_token={{ token }}&fields=id,name,effective_status,created_time and is returning something like this

`{'data': [{}, {}, {}], 'paging': { 'cursors': {'after': '<THE RANDOM KEY FOR AFTER>', 'before': '<THE RANDOM KEY FOR BEFORE>'}, 'next': <THE API ENDPOINT with after query param FOR THE NEXT PAGE> 'previous': <THE API ENDPOINT with before query param FOR THE PREV PAGE> }}`

So the limit query param is `limit` and that works well with the Data Provider pagination, but as pagination query params for the page I have `after` and `before` so not sure here how should I handle those other pagination inputs in the pagination REST API tool

![Screen Shot 2022-03-22 at 6 29 06 PM](https://user-images.githubusercontent.com/30072155/159586861-470bd72a-6825-44a2-91ea-147770327c08.png)

Thanks!

**Comment by melohagan** [comment, marked as answer]
Hey @lollococce 

I have recently added some documentation around the process of using pagination in REST API's which can be found here: https://docs.budibase.com/docs/rest-pagination

Your inputs look good there, the only thing that needs to change is the *Response body parameter name for cursor*.

I think for this API, the path would be: **paging.cursors.after**

There was a recent bug fix relating to response body pagination paths: https://github.com/Budibase/budibase/issues/5066
So the functionality may not be available until the next patch release.

**Reply by lollococce** [reply]
Thanks for the reply!!
